### PR TITLE
Visitor can only view part of premium article

### PR DESCRIPTION
**Pull request by johanperjulius1**
## This PR contains: 

- Acceptance test
- Adding premium as an attribute of article model
- Restrict premium content for visitor 

## [PR](https://www.pivotaltracker.com/n/projects/2468825/stories/175303495)

### CODE REVIEW

File: spec/requests/api/v1/articles/show_spec.rb [hunk 1]
```diff
@@ -40,4 +45,13 @@
       expect(response_json["error_message"]).to eq "Sorry we can not find that article"
     end
   end
+
+  describe "Visitor can only see part of premium content" do
```
Comment (emtalen) [inline spec/requests/api/v1/articles/show_spec.rb:51]: Please test that they can still see the full content of a none premium article

File: spec/requests/api/v1/articles/show_spec.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 RSpec.describe "GET /api/v1/articles", type: :request do
-  let!(:article) { create(:article) }
+  let!(:journalist) { create(:user, role: "journalist") }
```
Comment (emtalen) [inline spec/requests/api/v1/articles/show_spec.rb]: Why are you creating a journalist?

**Review by emtalen — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/articles/show_spec.rb (hunk 2 above)
Comment (johanperjulius1) [reply spec/requests/api/v1/articles/show_spec.rb]: We have an assoication that an article need a journalist. Then we need a journalist to create an article?

File: spec/requests/api/v1/articles/show_spec.rb (hunk 1 above)
Comment (johanperjulius1) [reply spec/requests/api/v1/articles/show_spec.rb:51]: Will do! Thanks!

**Review by emtalen — APPROVED** [review]

**Pull request merged**